test/boost/json_test.cc

```diff
@@ -54,3 +68,163 @@ BOOST_AUTO_TEST_CASE(test_parsing_map_from_null) {
+void test_format_range(const char* desc, Range x, std::unordered_set<std::string> expected_strings, std::string expected_delim = ",", bool expect_parenthesis = true) {
+    auto str = seastar::format("{}", x);
+    BOOST_TEST_MESSAGE(format("{}: {}", desc, str));
+
+    auto fmt_str = fmt::format("{}", x);
+    BOOST_REQUIRE_EQUAL(str, fmt_str);
+
+    size_t num_elements = expected_strings.size();
+
+    size_t paren_size = expect_parenthesis ? 2 : 0;
+    size_t min_size = paren_size + (x.empty() ? 0 : (num_elements - 1));
+    BOOST_REQUIRE_GE(str.size(), min_size);
+
+    std::string_view sv = str;
+    if (expect_parenthesis) {
+        verify_parenthesis(sv);
+        sv = sv.substr(1, sv.size() - 2);
+    }
+
+    bool first = true;
+    while (!expected_strings.empty()) {
+        sv = trim(sv);
+        if (!std::exchange(first, false)) {
+            sv = cmp_and_remove_prefix(sv, expected_delim);
+        }
+
+        for (auto it = expected_strings.begin(); it != expected_strings.end(); ++it) {
+            if (sv.starts_with(*it)) {
+                sv = cmp_and_remove_prefix(sv, *it);
+                expected_strings.erase(it);
+                break;
+            }
+        }
+    }
+
+    BOOST_REQUIRE(sv.empty());
+}
+
+} // namespace
+
+BOOST_AUTO_TEST_CASE(test_vector_format) {
+    auto vector = std::vector<int>({1, 2, 3});
+    test_format_range("vector", vector, std::vector<std::string>({"1", "2", "3"}));
+
+    auto array = std::array<int, 3>({1, 2, 3});
+    test_format_range("array", array, std::vector<std::string>({"1", "2", "3"}));
+
+    auto list = std::list<int>({1, 2, 3});
+    test_format_range("list", list, std::vector<std::string>({"1", "2", "3"}));
+
+    auto deque = std::deque<int>({1, 2, 3});
+    test_format_range("deque", deque, std::vector<std::string>({"1", "2", "3"}));
+
+    auto set = std::set<int>({1, 2, 3});
+    test_format_range("set", set, std::vector<std::string>({"1", "2", "3"}));
+
+    auto unordered_set = std::unordered_set<int>({1, 2, 3});
+    test_format_range("unordered_set", unordered_set, std::unordered_set<std::string>({"1", "2", "3"}));
+
+    auto map = std::map<int, std::string>({{1, "one"}, {2, "two"}, {3, "three"}});
+    test_format_range("map", map, std::vector<std::string>({"{1, one}", "{2, two}", "{3, three}"}));
+    test_format_range("map | boost::adaptors::map_keys", map | boost::adaptors::map_keys, std::vector<std::string>({"1", "2", "3"}));
+    test_format_range("map | boost::adaptors::map_values", map | boost::adaptors::map_values, std::vector<std::string>({"one", "two", "three"}));
+
+    auto unordered_map = std::unordered_map<int, std::string>({{1, "one"}, {2, "two"}, {3, "three"}});
+    // seastar has a specialized print function for unordered_map
+    // See https://github.com/scylladb/seastar/issues/1544
+    test_format_range("unordered_map", unordered_map, std::unordered_set<std::string>({"{1 -> one}", "{2 -> two}", "{3 -> three}"}));
```

Comment:
IMHO, the notation is kind of arbitrary, and it is not the same as how fmtlib formats `std::unordered_map`. do we rely on this formatting, like we do in dtest?

Reply:
I don't think we do, but it's kind of the convention now in scylla.
We use it to print e.g. option dictionaries (`std::unordered_map<sstring, sstring>`)

Reply:
thank you for confirming this. I was asking because I was actually thinking about dropping it, and replacing it with fmtlib. Will keep your answer in mind.